Sources/NetworkService/HTTPRequest.swift

```diff
@@ -0,0 +1,79 @@
+import Foundation
+
+/// Encapsulates an HTTP request with the necessary components.
```

Comment:
Can you please write some pseudocode (imagine we have implemented all necessary types), so I can understand where and how we can use the current `HTTPRequest` type?

It seems for the request we can use `URLRequest` object.

Reply:
I'd propose to remove the `HTTPRequest` type and rename/focus PR on the HTTPMethod/HTTPHeaders typealias to merge them quickly.

The following will be implemented by me.

Reply:
Fixed. I'll send pseudocode later in slack.